### PR TITLE
Ae test command initial

### PR DESCRIPTION
**Pull request by paetling**
WIP

### CODE REVIEW

**Comment by paetling** [comment]
review @thieman  This is sort of a shit show of a pr.   It is doing two distinct things.  First: it is adding a command that will recreate the testing compiled dusty_testing image.   Second: it is doing a small refactor of sync_repos_by_app_name and adding in sync_repos_by_lib_name.

File: dusty/systems/rsync/__init__.py [hunk 1]
```diff
@@ -54,11 +54,19 @@ def sync_repos(repos):
...
         repos.add(get_repo_of_app_or_library(app_name))
     sync_repos(repos)
+
+def sync_repos_by_lib_name(expanded_specs, lib_names):
+    repos = set()
+    for lib_name in lib_names:
+        for depends_lib_name in expanded_specs['libs'][lib_name].get('depends', {}).get('libs', []):
+            repos.add(get_repo_of_app_or_library(depends_lib_name))
+        repos.add(get_repo_of_app_or_library(lib_name))
+    sync_repos(repos)
```
Comment (thieman) [inline dusty/systems/rsync/__init__.py]: While you're here might as well rewrite these to remove all the duplication :shrug:

File: dusty/commands/tests.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    docker_client = get_docker_client()
+    expanded_specs = get_expanded_libs_specs()
+    if app_or_lib_name in expanded_specs['apps']:
+        volumes = get_app_volume_mounts(app_or_lib_name, expanded_specs)
+        spec = expanded_specs['apps'][app_or_lib_name]
+        sync_repos_by_app_name(expanded_specs, [app_or_lib_name])
+    elif app_or_lib_name in expanded_specs['libs']:
+        volumes = get_lib_volume_mounts(app_or_lib_name, expanded_specs)
+        spec = expanded_specs['libs'][app_or_lib_name]
+        sync_repos_by_lib_name(expanded_specs, [app_or_lib_name])
```
Comment (thieman) [inline dusty/commands/tests.py]: Good for now, specs class should make this a lot better

**Comment by thieman** [comment]
Couple things :bear:

**Pull request merged**